test/extended/util/client.go

```diff
@@ -735,14 +712,21 @@ func (c *CLI) Run(command string, flags ...string) *CLI {
 	if c.asPClusterKubeconf && !c.withoutNamespace {
 		FatalErr("you are doing something in ns of pcluster, please use WithoutNamespace and set ns in Args, for example, oc.AsPClusterKubeconf().WithoutNamespace().Run(\"get\").Args(\"pods\", \"-n\", \"pclusterns\").Output()")
 	}
-	if !c.withoutNamespace && c.Namespace() != "" {
-		nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
+
+	if !c.withoutNamespace {
+		isKCPCommand := c.isKCPCommand()
+		if isKCPCommand && c.currentWorkSpace.CurrentNameSpace != "" {
+			nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.currentWorkSpace.CurrentNameSpace))
+		} else if !isKCPCommand && c.Namespace() != "" {
+			nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
+		}
 	}
+
 	// TODO: Temp solution for parallelly ececute our test cases
 	// When https://github.com/kcp-dev/kcp/issues/1689 finished
 	// We could make it simply and just use the --kubeconfig instead of --server.
 	if !c.withoutWorkSpaceServer {
```

Comment:
I think we could move it before add the `--namespace=` as the ns should under ws.